policy/handler.go

```diff
@@ -218,3 +291,46 @@ func (h *Handler) updateHandler(current, next *Policy) {
 		h.ticker = time.NewTicker(next.EvaluationInterval)
 	}
 }
+
+// enforceCooldown blocks until the cooldown period has been reached, or the
+// handler has been instructed to exit. The boolean return details whether or
+// not the cooldown period passed without being interrupted.
+func (h *Handler) enforceCooldown(ctx context.Context, t time.Duration) bool {
+
+	// Log that cooldown is being enforced. This is very useful as cooldown
+	// blocks the ticker making this the only indication of cooldown to
+	// operators.
+	h.log.Debug("scaling policy has been placed into cooldown", "cooldown", t)
+
+	// Using a timer directly is mentioned to be more efficient than
+	// time.After() as long as we ensure to call Stop(). So setup a timer for
+	// use and defer the stop.
+	timer := time.NewTimer(t)
+	defer timer.Stop()
+
+	// Cooldown should not mean we miss other handler control signals. So wait
+	// on all the channels desired here.
+	select {
+	case <-timer.C:
+		return true
+	case <-ctx.Done():
+		return false
+	case <-h.doneCh:
+		return false
+
+	}
+}
+
+// isInCooldown is used to check whether the handler should be in cooldown or
+// not. This is used when performing a policy evaluation and allows the
+// autoscaler to enforce a cooldown on objects that have been scaled outside of
+// the autoscaler.
+func (h *Handler) isInCooldown(cd time.Duration, ts int64, lastEvent uint64) bool {
+	return int64(lastEvent)+cd.Nanoseconds() >= ts
+}
+
+// calculateRemainingCooldown calculates the remaining cooldown based on the
+// time since the last event.
+func (h *Handler) calculateRemainingCooldown(cd time.Duration, ts, lastEvent int64) time.Duration {
```

Comment:
combining `calculateRemainingCooldown` and `isInCooldown` into one method will help make sure that we don't accidentally mis-call the latter with a different `currentTime` than the former and a negative number. 

in fact, `isInCooldown == (calculateRemainingCooldown > 0)` (check my math), so they're redundant